pkg/kubectl/cmd/top_node.go

```diff
@@ -137,15 +151,40 @@ func (o *TopNodeOptions) Validate() error {
 
 func (o TopNodeOptions) RunTopNode() error {
 	var err error
-	selector := labels.Everything().String()
+	selector := labels.Everything()
 	if len(o.Selector) > 0 {
-		selector = o.Selector
+		selector, err = labels.Parse(o.Selector)
+		if err != nil {
+			return err
+		}
 	}
-	metrics, err := o.Client.GetNodeMetrics(o.ResourceName, selector)
+
+	apiGroups, err := o.DiscoveryClient.ServerGroups()
 	if err != nil {
 		return err
 	}
-	if len(metrics) == 0 {
+
+	metricsAPIAvailable := false
+	for _, group := range apiGroups.Groups {
+		if group.Name == metricsapi.GroupName {
+			metricsAPIAvailable = true
+		}
+	}
```

Comment:
We might want to check version as well

Reply:
Alternatively, we could try the metrics API client out of the box, and then retrying with the Heapster client if it returns a NotFound error.  I'm conflicted as to which is the correct approach

Reply:
Yes true, I'll make sure to add a version check.